### ISSUE TITLE
Add support for comma in function call and function declaration after the last argument (re-submitted)

### ISSUE THREAD

**Issue by jrencz** [eslint/espree#286]
This is a re-submission of #208

---

I'd like to give an example with AngularJS-centerd code, where function arguments have different meaning then the rest of JS world.

To ensure the diff to be as meaningful as possible I'd like to be able to declare dependency injection arguments line-by-line and make sure adding another won't modify the line history on any other line than the one I'm adding/modifying:

```javascript
angular.module('myModule', []).service('myService', function (
  dependencyService1,
  dependencyService2,
  dependencyService3
) {
  // service logic
})
```

(I'm not using array DI syntax for the sake of brevity)

If I want to add fourth dependency as the last argument (I'd probably do exactly that if I want my dependencies to be organised alphabetically or by some custom logic - whatever) I get this:

```javascript
angular.module('myModule', []).service('myService', function (
  dependencyService1,
  dependencyService2,
  dependencyService3, // changed line: added comma
  dependencyService4 // added line
) {
  // service logic
})
```

With last comma the example looks almost exactly the same:

```javascript
angular.module('myModule', []).service('myService', function (
  dependencyService1,
  dependencyService2,
  dependencyService3, // please notice the last comma
) {
  // service logic
})
```

```javascript
angular.module('myModule', []).service('myService', function (
  dependencyService1,
  dependencyService2,
  dependencyService3, // please notice: no change in this line
  dependencyService4, // added line: the change I've made is a one-liner, by diff is more meaningful!
) {
  // service logic
})
```

What I'd like to do is to be able to use last comma syntax and still use eslint, but it's not possible since code using commas after last function argument are not parsed by espree yet. 

```
Line 5: Unexpected token 
```

I know this syntax is not yet production ready, but for transpilers (namely: babel) [stripping this one comma is not a problem](https://babeljs.io/repl/#?experimental=true&evaluate=true&loose=false&spec=false&code=var%20a%20%3D%20function%20Foo(%0A%20%20a%2C%0A%20%20b%2C%0A%20%20c%2C%0A)%20%7B%7D%3B)

**Comment by ljharb** [comment]
Repeating my comment from https://github.com/eslint/espree/issues/208#issuecomment-235700862:
This proposal advanced today to stage 4 at the July TC39 meeting (tc39/proposals@6826a19), which means it's past time to support it :-)

**Comment by nzakas** [comment]
It actually means it's the start of time when we can support it. :) Remember, we don't do anything until it's stage 4.

I'm waiting to hear back on what Acorn's position is on this: https://github.com/ternjs/acorn/issues/441

**Comment by nzakas** [comment]
I started work on this and trailing commas turned out to be pretty easy. This is the Acorn fork I'm using: https://github.com/nzakas/acorn/tree/issue441

Just need to wait to hear back on Acorn's position.

**Comment by SimenB** [comment]
This was merged into acorn a week ago 😄 ternjs/acorn#461

**Comment by nzakas** [comment]
Work to integrate Acorn is happening here: https://github.com/eslint/espree/issues/287
